### PR TITLE
Fix two small issues

### PR DESCRIPTION
**Pull request by pjrobertson**
These are some small issues I came across whilst setting up/testing the project. I will add to this PR if I find any more. For bigger issues/features, I'll create new issues.

### CODE REVIEW

File: src/auto_archiver/formatters/templates/macros.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {% for url in m.urls %}
 {% if url | length == 0 %}
 No URL available for {{ m.key }}.
-{% elif 'http' in url %}
+{% elif 'http://' in url or 'https://' in url or url.startswith('/') %}
```
Comment (msramalho) [inline src/auto_archiver/formatters/templates/macros.html:6]: good catch!

File: src/auto_archiver/archivers/twitter_archiver.py [hunk 2]
```diff
@@ -147,10 +147,16 @@ def download_yt_dlp(self, item: Metadata, url: str, tweet_id: str) -> Union[Meta
         tie = TwitterIE(downloader)
         tweet = tie._extract_status(tweet_id)
         result = Metadata()
+        try:
+            timestamp = datetime.strptime(tweet["created_at"], "%a %b %d %H:%M:%S %z %Y")
+        except Exception as ex:
+            logger.warning(f"Failed to get timestamp: {type(ex).__name__} occurred. args: {ex.args}")
+            return False
```
Comment (msramalho) [inline src/auto_archiver/archivers/twitter_archiver.py:154]: assumption here that if the datetime parse fails the response was actually broken, is it so?

I see this is the previous scenario but with more granular logging.

Comment (pjrobertson) [reply src/auto_archiver/archivers/twitter_archiver.py:154]: Yes, I haven't changed the functionality, just made clearer logging (I think I got an error on this at one point, which is why I changed it for clearer logging). Something for us to discuss going forwards. I'll flag this on the project so I can discuss with @erinhmclark - i.e.: logging and how to deal with potentially broken/incomplete/unexpected responses (is an 'unexpected' response better than nothing?)

Comment (pjrobertson) [reply src/auto_archiver/archivers/twitter_archiver.py:154]: OK, so running on an invalid tweet gives the following response from youtubedlp

```
(Pdb) !tweet
{'user': {}, 'card': {}, 'quoted_status': {}, 'retweeted_status': {}}
```

We need some kind of better validation to make sure the tweet is valid. Right now it'll fail because the timestamp isn't parsed correctly (it's empty) but that's a bit brittle. Will add it to this PR

**Comment by pjrobertson** [comment]
OK, improved logging and pushed another commit along with unit tests to clarify the edge case where parsing the timestamp was failing (it's actually because the tweet didn't exist). Good catch, thanks!

[WIP] tag removed. Should be good to go

**Comment by pjrobertson** [comment]
Note: this now also includes a fix for the 'download_syndication' method, which was previously failing. Follows the discussion at https://github.com/JustAnotherArchivist/snscrape/issues/996#issuecomment-2211358215

File: tests/archivers/test_twitter_archiver.py [hunk 3]
```diff
@@ -70,8 +69,42 @@ def test_youtube_dlp_archiver(self):
...
+        )
 
+    def test_download_nonexistend_tweet(self):
+        # this tweet does not exist
+        url = "https://x.com/Bellingcat/status/17197025860711058"
+        response = self.archiver.download(self.create_item(url))
+        self.assertFalse(response)
+
+    def test_download_malformed_tweetid(self):
+        # this tweet does not exist
```
[the remark this answers is not in the document]
Comment (pjrobertson) [reply tests/archivers/test_twitter_archiver.py:104]: Yep, I'm switching over to pytest in the unittest PR, we can do that there
